### PR TITLE
Adds Support for Control Plane TLS

### PR DESCRIPTION
**Pull request by danehans**
- Adds a package for generating control plane auth certs.
- Adds a const to the EG config pkg to represent the default "envoy-gateway-system" ns.
- Updates `ObjectName()` method to get/set the default name of the ProxyInfra type.

fixes: #70 

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

**Comment by LukeShu** [comment]
To communicate: I haven't been looking at this because it is failing CI.

**Comment by danehans** [comment]
@LukeShu I've been side tracked with other PR. I marked the PR as WIP.

File: internal/crypto/certgen_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	expiry := now.Add(24 * 365 * time.Hour)
+
+	caCert, caKey, err := newCA("envoy-gateway", expiry)
+	require.NoErrorf(t, err, "Failed to generate CA cert")
+
+	egCertReq := &certificateRequest{
+		caCertPEM:  caCert,
+		caKeyPEM:   caKey,
+		expiry:     expiry,
+		commonName: "envoy-gateway",
```
Comment (lizan) [inline internal/crypto/certgen_test.go:65]: Common names are deprecated in RFC 2818 (in 2000) in favor of SANs. Major softwares (Golang, Chrome etc) stopped supporting them, so we should just issue certs without common names.

**Comment by danehans** [comment]
xref #404 will causes this PR to support a TLS secret per managed Envoy deployment.

File: internal/crypto/certgen.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+		DNSNames: request.altNames,
+	}
+	newCert, err := x509.CreateCertificate(rand.Reader, template, caCert, &newKey.PublicKey, caKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	newKeyPEM := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(newKey),
```
Comment (LukeShu) [inline internal/crypto/certgen.go:204]: Getting in to bike-shed territory, but any reason to not use `"PRIVATE KEY"` and `x509.MarshalPKCS8PrivateKey`? It'd have fewer things to change if we ever want to switch away from RSA.

**Review by LukeShu — APPROVED** [review]
Some nit-picks, but on the whole LGTM.

**Comment by LukeShu** [comment]
If anyone cares, here's some very similar code in Emissary: https://github.com/emissary-ingress/emissary/blob/master/cmd/apiext/ca.go

**Pull request merged**